client/landing/site.landing/coffee/core/utils.coffee

```diff
@@ -60,31 +60,40 @@ module.exports = utils = {
 
   getGroupNameFromLocation: (hostname) ->
 
-    { hostname }     = location  unless hostname
-    ipV4Pattern      = '(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.xip\.io$'
-    subDomainPattern = '[A-Za-z0-9_]{1}[A-Za-z0-9_-]+'
-    kodingDomains    = '(?:dev|sandbox|latest|prod)'
-
-    # Base Domain may include port information which we don't need ~GG
-    baseDomain       = (kd.config.domains.main.split ':')[0]
-    baseDomain       = "#{baseDomain}".replace /\./g, '\\.'
-
-    # e.g. [teamName.]dev|sandbox|latest|prod.koding.com
-    teamPattern = if ///#{kodingDomains}\.#{baseDomain}$///.test hostname
-    then ///(?:^(#{subDomainPattern})\.)?#{kodingDomains}\.#{baseDomain}$///
-    # e.g. [teamName.]koding.com
-    else if ///#{baseDomain}$///.test hostname
-    then ///(?:^(#{subDomainPattern})\.)?#{baseDomain}$///
-    # e.g. [teamName.]<teamMember>.koding.team
-    else if /koding\.team$/.test hostname
-    then ///(?:^(#{subDomainPattern})\.)?(?:#{subDomainPattern}\.)koding\.team$///
-    # e.g. [teamName.]<vm-ip>.xip.io
-    else ///(?:^(#{subDomainPattern})\.)?#{ipV4Pattern}///
-
-    matches  = hostname.match teamPattern
-    teamName = matches?[1] or 'koding'
+    { hostname } = location  unless hostname
+    basePattern  = (kd.config.domains.main.split ':')[0].replace /\./g, '\\.'
 
-    return teamName
+    predefinedPatterns = [
+      # <team-domain>.<(dev|sandbox|latest|prod)>.koding.com
+      { predict: "(dev|sandbox|latest|prod)\.#{basePattern}" }
+
+      # <team-domain>.koding.com
+      # <team-domain>.<username>.koding.team
+      { predict: "#{basePattern}" }
+
+      # <team-domain>.<IPv4>.xip.io
+      # IPV4: A.B.C.D
+      {
+        predict: 'xip\.io'
```

Comment:
I can't think of any practical way to do that. It is much easier to add a new rule for a target service. Some expect host address to be constructed with dashes instead of dots etc.